apis/keda/v1alpha1/scaledobject_webhook.go

```diff
@@ -150,31 +148,29 @@ func isRemovingFinalizer(so *ScaledObject, old kruntime.Object) bool {
 func validateWorkload(so *ScaledObject, action string, dryRun bool) (admission.Warnings, error) {
 	metricscollector.RecordScaledObjectValidatingTotal(so.Namespace, action)
 
-	verifyFunctions := []func(*ScaledObject, string, bool) error{
-		verifyCPUMemoryScalers,
-		verifyScaledObjects,
-		verifyHpas,
-		verifyReplicaCount,
-		verifyFallback,
+	verifyFunctions := map[string]func(*ScaledObject, string, bool) error{
+		"verifyCPUMemoryScalers": verifyCPUMemoryScalers,
+		"verifyScaledObjects":    verifyScaledObjects,
+		"verifyHpas":             verifyHpas,
+		"verifyReplicaCount":     verifyReplicaCount,
+		"verifyFallback":         verifyFallback,
 	}
```

Comment:
@JorTurFer, @zroubalik do you know if the order in which the verify functions are called is relevant?

Reply:
in theory, the order is totally irrelevant and we could parallelize them if we want, as they check multiple things. And thinking about my last point, maybe we should validate all of them before giving an error and return all the validation errors at once instead of just failing fast